.github/workflows/release.yml

```diff
@@ -0,0 +1,48 @@
+---
+name: release
+
+on:
+  workflow_dispatch: {}
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: set-up-jdk
+        uses: actions/setup-java@v1
+        with:
+          java-version: "11"
+          server-id: ossrh
+          settings-path: ${{ github.workspace }}
+          server-username: MAVEN_USERNAME
+          server-password: MAVEN_PASSWORD
+          gpg-private-key: ${{ secrets.MAVEN_GPG_PRIVATE_KEY }}
+          gpg-passphrase: MAVEN_GPG_PASSPHRASE
```

Comment:
Should the username, password, and gpg_passphrase also be coming from `secrets`?

Reply:
They are, they're loaded in through `env` vars in the next step. This is setting up the env variable name for resolution resolution in `settings.xml` in the next step below. They're just named a little different, though it can change. The names are setup like this to show the difference between how the `settings.xml` var of `MAVEN_USERNAME` is setup vs. how it's resolved against the `env` var of `OSSRH_USERNAME` passed in through `secrets` in the next step 👍 

I'm under the impression it's setup this way to reduce risks with credentials being persisted to disk on the action runner--  see the `setup-java` example for info: https://github.com/actions/setup-java#publishing-using-apache-maven

Reply:
Got it thanks for explaining